src/models/relations-container.ts

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Relations } from "./relations";
+import { EventType, RelationType } from "../@types/event";
+import { EventStatus, MatrixEvent, MatrixEventEvent } from "./event";
+import { EventTimelineSet } from "./event-timeline-set";
+import { MatrixClient } from "../client";
+import { Room } from "./room";
+
+export class RelationsContainer {
+    // A tree of objects to access a set of relations for an event, as in:
+    // this.relations[relatesToEventId][relationType][relationEventType]
+    private relations: {
+        [relatesToEventId: string]: {
+            [relationType: RelationType | string]: {
+                [eventType: EventType | string]: Relations;
+            };
+        };
+    } = {};
+
+    constructor(private readonly client: MatrixClient, private readonly room?: Room) {
+    }
+
+    /**
+     * Get a collection of relations to a given event in this timeline set.
```

Comment:
There's a reason why the spec decided to abandon all hope of documenting what a "relation event", "related event", and "relatable event" are: it gets quite complicated. 


Having reviewed this PR, I think it's quite important that we match the spec on this as it's nearly impossible to understand what event is being talked about with the current overlap in vocabulary. The spec uses "parent" and "child" to denote the relationship between the events, described here: https://github.com/matrix-org/matrix-spec/pull/1062